overlord/hookstate/ctlcmd/list_connected_test.go

```diff
@@ -0,0 +1,95 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package ctlcmd_test
+
+import (
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/overlord/hookstate"
+	"github.com/snapcore/snapd/overlord/hookstate/ctlcmd"
+	"github.com/snapcore/snapd/overlord/hookstate/hooktest"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type listConnectedSuite struct {
+	testutil.BaseTest
+	st          *state.State
+	mockHandler *hooktest.MockHandler
+}
+
+var _ = Suite(&listConnectedSuite{})
+
+func (s *listConnectedSuite) SetUpTest(c *C) {
+	s.BaseTest.SetUpTest(c)
+	dirs.SetRootDir(c.MkDir())
+	s.AddCleanup(func() { dirs.SetRootDir("/") })
+	s.st = state.New(nil)
+	s.mockHandler = hooktest.NewMockHandler()
+}
+
+func (s *listConnectedSuite) TestListConnected(c *C) {
+	s.st.Lock()
+	defer s.st.Unlock()
+
+	task := s.st.NewTask("test-task", "my test task")
+	setup := &hookstate.HookSetup{Snap: "snap1", Revision: snap.R(1), Hook: "test-hook"}
+
+	mockContext, err := hookstate.NewContext(task, s.st, setup, s.mockHandler, "")
+	c.Check(err, IsNil)
+
+	mockInstalledSnap(c, s.st, `name: snap1
+plugs:
+  plug1a:
+    interface: x11
+  plug1b:
+    interface: x11
+slots:
+  slot1a:
+    interface: x11
+  slot1b:
+    interface: x11`, "")
+	mockInstalledSnap(c, s.st, `name: snap2
+plugs:
+  plug2a:
+    interface: x11
+  plug2b:
+    interface: x11
+slots:
+  slot2a:
+    interface: x11
+  slot2b:
+    interface: x11`, "")
+	s.st.Set("conns", map[string]interface{}{
+		"snap1:plug1a snap2:slot2a": map[string]interface{}{},
+		"snap2:plug2a snap1:slot1a": map[string]interface{}{},
+	})
```

Comment:
w.r.t. an earlier comment, it might be worth adding a third test snap so you can have a slot with multiple connected plugs.